Source/UI/Home/HomeView.swift

```diff
@@ -12,14 +12,17 @@ import Logger
 import SwiftUI
 
 struct HomeView: View, HelpDrawerHost {
-    @ObservedObject
-    private var dataSource: FeedStrategyMessageList
-
-    init(dataSource: FeedStrategyMessageList, helpDrawerState: HelpDrawerState) {
-        self.dataSource = dataSource
+    init(helpDrawerState: HelpDrawerState, bot: Bot) {
```

Comment:
I think `HomeView` has two references to a `Bot` now. One is passed in the initializer, and one is an `@EnvironmentObject`, and theoretically they could be different. Maybe in this initializer we can use use the bot from the environment?

Reply:
They are different, one is botRepository and the other is Bot. To be honest, I don't like this solution either, botRepository is meant to be transitory until we work on #981 and can hopefully change that there.

Reply:
Could we just have a computed property named `bot` that reads the `botRepository` from the environment?

Reply:
Unfortunately not, I wish I could, I had a hard time trying to figure it out and that's why I did it in this way. The environment objects are not set and cannot be used in the initializers.

Reply:
ah ok. I think I have run into that before as well.